packages/contracts/contracts/test/ValidatorManager.t.sol

```diff
@@ -1038,3 +1039,96 @@ contract ValidatorManagerTest is ValidatorSystemUpgrade_Initializer {
         assertEq(valMgr.activatedValidatorTotalWeight(), totalWeight + 10);
     }
 }
+
+contract ValidatorManager_MptTransition_Test is ValidatorSystemUpgrade_Initializer {
+    MockL2OutputOracle mockOracle;
+    MockValidatorManager mockValMgr;
+    function setUp() public override {
+        // value greater than terminateOutputIndex(3)
+        mptFirstOutputIndex = 5;
+
+        super.setUp();
+
+        address oracleAddress = address(oracle);
+        MockL2OutputOracle mockOracleImpl = new MockL2OutputOracle(
+            pool,
+            valMgr,
+            address(colosseum),
+            submissionInterval,
+            l2BlockTime,
+            startingBlockNumber,
+            startingTimestamp,
+            finalizationPeriodSeconds
+        );
+        vm.prank(multisig);
+        Proxy(payable(oracleAddress)).upgradeTo(address(mockOracleImpl));
+        mockOracle = MockL2OutputOracle(oracleAddress);
+
+        address valMgrAddress = address(valMgr);
+        MockValidatorManager mockValMgrImpl = new MockValidatorManager(constructorParams);
+        vm.prank(multisig);
+        Proxy(payable(valMgrAddress)).upgradeTo(address(mockValMgrImpl));
+        mockValMgr = MockValidatorManager(valMgrAddress);
+
+        // Submit until terminateOutputIndex and set next output index to be finalized after it
+        vm.prank(trusted);
+        pool.deposit{ value: trusted.balance }();
+        for (uint256 i = oracle.nextOutputIndex(); i <= terminateOutputIndex; i++) {
+            _submitL2OutputV1();
+        }
+
+        vm.warp(oracle.finalizedAt(terminateOutputIndex));
+        mockOracle.mockSetNextFinalizeOutputIndex(terminateOutputIndex + 1);
+
+        // Since the first output of V2 is always for trusted validator,
+        // so for accurate test, we should pass the first V2 output
+        uint128 assets = minActivateAmount;
+        _registerValidator(asserter, assets);
+        _registerValidator(trusted, assets);
+        _submitL2OutputV2(false);
+
+        vm.warp(oracle.finalizedAt(terminateOutputIndex + 1));
+        mockOracle.mockSetNextFinalizeOutputIndex(terminateOutputIndex + 2);
+    }
+
+    function test_submitL2Output_mptFirstOutput_privateRound_trustedValidator_succeeds() public {
+        assertEq(valMgr.nextValidator(), trusted);
+        _submitL2OutputV2(false);
+    }
+
+    function test_submitL2Output_mptFirstOutput_publicRound_trustedValidator_succeeds() public {
+        // Warp to public round since trusted validator is always selected for the priority validator
```

Comment:
Since asserter and trusted already registered before the first V2 output was submitted, so asserter can be selected for the priority validator proportionaly
```suggestion
        // Warp to public round
```